device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -108,33 +106,22 @@
             throw new IllegalArgumentException("Config cannot be null.");
         }
 
-        /* Codes_SRS_DEVICE_IO_21_001: [The constructor shall store the provided protocol and config information.] */
-        deviceClientConfigs.add(config);
-        this.config = config;
-        this.protocol = this.config.getProtocol();
-
-        /* Codes_SRS_DEVICE_IO_21_037: [The constructor shall initialize the `sendPeriodInMilliseconds` with default value of 10 milliseconds.] */
-        this.sendPeriodInMilliseconds = sendPeriodInMilliseconds;
-        /* Codes_SRS_DEVICE_IO_21_038: [The constructor shall initialize the `receivePeriodInMilliseconds` with default value of each protocol.] */
-        this.receivePeriodInMilliseconds = receivePeriodInMilliseconds;
+        this.protocol = config.getProtocol();
 
-        /* Codes_SRS_DEVICE_IO_21_006: [The constructor shall set the `state` as `DISCONNECTED`.] */
-        this.state = IotHubClientState.CLOSED;
-
-        if (protocol == IotHubClientProtocol.AMQPS_WS || protocol == IotHubClientProtocol.MQTT_WS)
+        if (this.protocol == IotHubClientProtocol.AMQPS_WS || this.protocol == IotHubClientProtocol.MQTT_WS)
         {
-            this.config.setUseWebsocket(true);
+            config.setUseWebsocket(true);
         }
 
-        this.transport = new IotHubTransport(config);
+        /* Codes_SRS_DEVICE_IO_21_001: [The constructor shall store the provided protocol and config information.] */
+        this.deviceClientConfigs.add(config);
+
```

Comment:
Would Line #117 override with defaults and #113 would never be set?

Reply:
I'm not sure I follow what you mean here. The config passed in will always set the appropriate websocket settings, and then it will save that config to our list

---
In reply to: [338271300](https://github.com/Azure/azure-iot-sdk-java/pull/608#discussion_r338271300) [](ancestors = 338271300)